packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -811,6 +813,34 @@ describe("SharedTree", () => {
 			validateTree(tree1, [expectedState]);
 			validateTree(tree2, [expectedState]);
 		});
+		it("rebases stashed ops with prior state present", async () => {
+			const provider = await TestTreeProvider.create(2);
+			insert(provider.trees[0], 0, "a");
+			await provider.ensureSynchronized();
+
+			const pausedContainer = provider.containers[0];
+			const url = await pausedContainer.getAbsoluteUrl("");
+			const pausedTree = provider.trees[0];
+			await provider.opProcessingController.pauseProcessing(pausedContainer);
+			insert(pausedTree, 1, "b");
+			insert(pausedTree, 2, "c");
+			const pendingOps = pausedContainer.closeAndGetPendingLocalState();
+			provider.opProcessingController.resumeProcessing();
+
+			const otherLoadedTree = provider.trees[1];
+			insert(otherLoadedTree, 0, "d");
+			await provider.ensureSynchronized();
+
+			const loader = provider.makeTestLoader();
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Why disable our linter rule that code should make assumptions about data like this when its simpler and less code (and require less detailed review) if you just check for null/undefined explicitly?

You can replace this line with assert(url !== undefined) or adjust the line above to `const url = await pausedContainer.getAbsoluteUrl("") ?? fail("didn't get url");`

Either of these would be preferred over suppressing the linter.

As a general rule, when disabling the linter, you should include a comment explaining why the documented reason for that linter rule existing does not apply to this case. (ex: the docs for this one are here, and it suggest making the code more typesafe instead which is what I suggested above: https://typescript-eslint.io/rules/no-non-null-assertion/ . If for some reason that's not possible or adds undesired overhead, suppressing the linter with an explanation of why might make sense, but there really isn't a justification for it in this case)

Reply:
adjusted to `const url = await pausedContainer.getAbsoluteUrl("") ?? fail("didn't get url");`